lib/dotcom_web/live/trip_planner.ex

```diff
@@ -426,16 +426,24 @@ defmodule DotcomWeb.Live.TripPlanner do
   #
   # - Update the input form state with the new changeset
   # - Update the map state with the new pins
-  # - Set the results state to loading
-  # - Start an async operation to get the itinerary groups
+  # - If the changeset is valid:
+  #   - Set the results state to loading
+  #   - Start an async operation to get the itinerary groups
   defp submit_changeset(socket, changeset) do
     new_changeset = Map.put(changeset, :action, :submit)
 
-    socket
-    |> assign(:input_form, Map.put(@state.input_form, :changeset, new_changeset))
-    |> assign(:map, Map.put(@state.map, :pins, input_form_to_pins(new_changeset)))
-    |> assign(:results, Map.put(@state.results, :loading?, true))
-    |> start_async("get_itinerary_groups", fn -> get_itinerary_groups(new_changeset) end)
+    new_socket =
+      socket
+      |> assign(:input_form, Map.put(@state.input_form, :changeset, new_changeset))
+      |> assign(:map, Map.put(@state.map, :pins, input_form_to_pins(new_changeset)))
+
+    if new_changeset.valid? do
+      new_socket
+      |> assign(:results, Map.put(@state.results, :loading?, true))
+      |> start_async("get_itinerary_groups", fn -> get_itinerary_groups(new_changeset) end)
+    else
+      new_socket
```

Comment:
I think this needs to explicitly clear the results if they're already set. Something like 👇 maybe?
```suggestion
      new_socket
      |> assign(:results, @state.results)
```

Reply:
> I think this needs to explicitly clear the results if they're already set. Something like 👇 maybe?

That will make the existing results disappear even if the new form submission isn't valid. Is that what we want?

Reply:
Haha great question! I _think_ the answer is yes, but let's ask Chase.